driver/pelican/types/scheduleSet.go

```diff
@@ -0,0 +1,88 @@
+package types
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/parnurzeal/gorequest"
+	rrule "github.com/teambition/rrule-go"
+)
+
+func (pel *Pelican) SetSchedule(newSchedule *ThermostatSchedule) error {
+	// Retrieve thermostat's latest schedule for extra block deletion purposes
+	currentSchedule, currentErr := pel.GetSchedule()
+	if currentErr != nil {
+		return fmt.Errorf("Error retrieving thermostat %v's current schedule: %v", pel.Name, currentErr)
+	}
+
+	for day, blocks := range newSchedule.DaySchedules {
+		// Delete Unnecessary Blocks + Error Checking
+		currentBlockCount := len(currentSchedule.DaySchedules[day])
+		requiredBlockCount := len(blocks)
+		for requiredBlockCount < currentBlockCount {
+			respDelete, _, errsDelete := gorequest.New().Get(pel.target).
+				Param("username", pel.username).
+				Param("password", pel.password).
+				Param("request", "set").
+				Param("object", "thermostatSchedule").
+				Param("selection", fmt.Sprintf("name:%s;dayOfWeek:%s;setTime:%v;", pel.Name, day, currentBlockCount)).
+				Param("value", "delete").
+				End()
+
+			if errsDelete != nil {
+				return fmt.Errorf("Error deleting thermostat schedule settings on day %v: %v\n", day, errsDelete)
+			}
+			var resultDelete apiResult
+			decDelete := xml.NewDecoder(respDelete.Body)
+			if errDecodeDelete := decDelete.Decode(&resultDelete); errDecodeDelete != nil {
+				return fmt.Errorf("Failed to decode thermostat schedule delete response XML: %v", errDecodeDelete)
+			}
+			if resultDelete.Success == 0 {
+				return fmt.Errorf("Error deleting thermostat schedule settings on day %v: %v\n", day, resultDelete.Message)
+			}
+			defer respDelete.Body.Close()
+			currentBlockCount -= 1
+		}
+
+		// Construct and Set New Schedule Settings by Block
+		for index, block := range blocks {
+			var value string = ""
+			value += fmt.Sprintf("coolSetting:%.0f;", block.CoolSetting)
+			value += fmt.Sprintf("heatSetting:%.0f", block.HeatSetting)
+			value += fmt.Sprintf("system:%s", block.System)
+
+			// Convert Time to Pelican's Timezone
+			time, timeErr := rrule.StrToRRule(block.Time)
+			if timeErr != nil {
+				return fmt.Errorf("Error converting time string %v to RRule format: %v\n", block.Time, timeErr)
+			}
+			timeLocal := time.OrigOptions.Dtstart.In(pel.timezone)
+			value += fmt.Sprintf("startTime:%s;", fmt.Sprintf("%v:%v", timeLocal.Hour(), timeLocal.Minute()))
```

Comment:
Let's use Go's `Format` function from the `time` package to do this. I know it's roughly equivalent to what you have, but that should make it more obvious what's going on here.

Reply:
Gotcha, made the change!